pkg/providers/imagefamily/nodeimage_test.go

```diff
@@ -97,32 +108,52 @@ var _ = Describe("NodeImageProvider tests", func() {
-			nodeClass.StatusConditions().SetFalse(v1beta1.ConditionTypeKubernetesVersionReady, "KubernetesVersionFalseForTesting", "tesitng false kubernetes version status")
+			nodeClass.StatusConditions().SetFalse(v1beta1.ConditionTypeKubernetesVersionReady, "KubernetesVersionFalseForTesting", "testing false kubernetes version status")
 			_, err := nodeImageProvider.List(ctx, nodeClass)
 			Expect(err).To(HaveOccurred())
-			Expect(err).To(Equal(fmt.Errorf("NodeClass condition %s, is in Ready=%s, %s", v1beta1.ConditionTypeKubernetesVersionReady, "False", "tesitng false kubernetes version status")))
+			Expect(err).To(Equal(fmt.Errorf("NodeClass condition %s, is in Ready=%s, %s", v1beta1.ConditionTypeKubernetesVersionReady, "False", "testing false kubernetes version status")))
 		})
 
 		It("should match expected images for Ubuntu2204", func() {
 			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
 			Expect(err).ToNot(HaveOccurred())
-			Expect(foundImages).To(ContainElements(getExpectedTestCIGImages(*nodeClass.Spec.ImageFamily, cigImageVersion)))
+			Expect(foundImages).To(ContainElements(getExpectedTestCIGImages(*nodeClass.Spec.ImageFamily, cigImageVersion, kubernetesVersion)))
 		})
 
-		It("should match expected images for AzureLinux", func() {
-			imageFamily := v1beta1.AzureLinuxImageFamily
-			nodeClass.Spec.ImageFamily = &imageFamily
+		// This test changes depending on the Kubernetes version, in effect making the following version-specific tests unnecessary.
+		// They are still kept for clarity and to ensure that the behavior is explicitly tested.
+		It("should match expected images for AzureLinux, depending on the Kubernetes version", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
 
 			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
 			Expect(err).ToNot(HaveOccurred())
-			Expect(foundImages).To(ContainElements(getExpectedTestCIGImages(*nodeClass.Spec.ImageFamily, cigImageVersion)))
+			Expect(foundImages).To(ContainElements(getExpectedTestCIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, kubernetesVersion)))
+		})
+
+		It("should match expected images for AzureLinux with version < 1.32", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
+			nodeClass.Status.KubernetesVersion = "1.31.0"
+
+			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(foundImages).To(ContainElements(getExpectedTestCIGImages(*nodeClass.Spec.ImageFamily, cigImageVersion, "1.31.0")))
+		})
+
+		It("should match expected images for AzureLinux with version >= 1.32", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
+			nodeClass.Status.KubernetesVersion = "1.32.0"
+
+			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(foundImages).To(ContainElements(getExpectedTestCIGImages(*nodeClass.Spec.ImageFamily, cigImageVersion, "1.32.0")))
```

Comment:
With these being explicate version tests, do you want to directly check for at least one exact ID in the foundImages, rather than just calling `getExpectedTestCIGImages`?

Just since as you mentioned above the tests themselves check over the different versions, and this is more for clarity/local testing one one version.

Reply:
Updated for readability, PTAL

Reply:
lgtm